lib/service/dataLink.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const kinveyCompletionHandler = require('./kinveyCompletionHandler');
+const kinveyErrors = require('kinvey-datalink-errors');
+const notImplementedHandler = require('./notImplementedHandler');
+
+let serviceObject = '';
```

Comment:
This should be avoided.  If you follow the pattern mentioned earlier of declaring the function separate from the exports, this isn't needed.